kvbc/include/categorization/updates.h

```diff
@@ -143,26 +143,27 @@ struct MerkleUpdates {
 // Updates contains a list of updates for different categories.
 // Note: Only a single `SharedKeyValueUpdates` is supported per block.
 struct Updates {
-  void addMerkleUpdate(std::string category_id, MerkleUpdates&& updates) {
-    if (const auto [itr, inserted] = category_updates_.insert({category_id, std::move(updates.data_)}); inserted) {
+  void add(std::string&& category_id, MerkleUpdates&& updates) {
+    if (const auto [itr, inserted] = category_updates_.try_emplace(std::move(category_id), std::move(updates.data_));
+        !inserted) {
       *itr;  // disable unused variable
       throw std::logic_error{std::string("Only one update for category is allowed. type: Merkle, category: ") +
                              category_id};
     }
   }
 
-  void addKeyValueUpdates(std::string category_id, KeyValueUpdates&& updates) {
-    if (const auto [itr, inserted] = category_updates_.insert({category_id, std::move(updates.data_)}); inserted) {
+  void add(std::string&& category_id, KeyValueUpdates&& updates) {
+    if (const auto [itr, inserted] = category_updates_.try_emplace(std::move(category_id), std::move(updates.data_));
+        !inserted) {
       *itr;  // disable unused variable
```

Comment:
Ah, I wish one could use `[[maybe_unused]]` with structured bindings. Maybe we can use the `(void)itr` hack to avoid any possible runtime cost from `*`. Not that the compiler will do something on that line in O3...